zozulyavladimir/lab1-BasicIO/user.c

```diff
@@ -0,0 +1,297 @@
+/******************************************************************************/
+/* Files to Include                                                           */
+/******************************************************************************/
+
+#ifdef __XC32
+#include <xc.h>          /* Defines special function registers, CP0 regs  */
+#endif
+
+#include <stdint.h>          /* For uint32_t definition                       */
+#include <stdbool.h>         /* For true/false definition                     */
+#include "user.h"            /* variables/params used by user.c               */
+
+/******************************************************************************/
+/* User Functions                                                             */
+
+/******************************************************************************/
+
+void InitApp(void) {
+    /* Setup analog functionality and port direction */
+
+    // LED output
+    // Disable analog mode for G6
+<<<<<<< HEAD
+    ANSELG |= ~(BIT_6);
```

Comment:
At init (only) it's better to use direct assignments. This way, it becomes possible to avoid errors related to after-reset register values and to use less instructions, like:
`ANSELG = BIT_15 | BIT_11 | BIT_6;`

Also, the `A |= ~(B)` is invalid. There are 3 common types of bit operations used in embedded:
`A |= (B);` -- Set bit
`A &= ~(B);` -- Unset bit
`A ^= (B);` -- Flip bit
Other combinations are **also possible**, but numerous `A |= ~(B)` look erroneous